src/ActionModal/components/SendModal.vue

```diff
@@ -276,12 +274,15 @@ export default {
       this.sending = false
     },
     setMaxAmount() {
-      if (this.network.startsWith(`terra`)) {
-        const terraTax = 0.008
-        this.amount = maxDecimals(
-          this.selectedBalance.amount / (1 + terraTax),
-          6 // TODO get precision fro API
+      if (
+        this.network.startsWith(`terra`) &&
+        this.selectedBalance.denom !== `LUNA`
+      ) {
+        const terraTax = Math.min(
+          Number(this.selectedBalance.amount) * TERRA_TAX_RATE,
+          TERRA_TAX_CAP
         )
+        this.amount = this.selectedBalance.amount - terraTax
       } else {
         this.amount = this.selectedBalance.amount
       }
```

Comment:
```suggestion
      let chainAppliedFees = 0
      if (
        this.network.startsWith(`terra`) &&
        this.selectedBalance.denom !== `LUNA`
      ) {
        const terraTax = Math.min(
          Number(this.selectedBalance.amount) * TERRA_TAX_RATE,
          TERRA_TAX_CAP
        )
        chainAppliedFees = terraTax
      }
      this.amount = this.selectedBalance.amount - chainAppliedFees
```

Reply:
I have refactored everything and now terraTax is a prop coming from SendModal. Since there is no point or use case for the other actions... except perhaps claim rewards when there is only alt-tokens to claim 🤔 

I'm looking into it